scipy/spatial/transform/rotation.py

```diff
@@ -281,3 +282,147 @@ def as_rotvec(self):
+        seq = seq.lower()
+        num_rotations = angles.shape[0]
+        quat = np.empty((num_rotations, 4, len(seq)))
+
+        for idx, axis in enumerate(seq):
+            quat[:, :, idx] = cls._make_elementary_quat(axis, angles[:, idx])
+
+        # Initialize result to identity quaternions
+        result = np.zeros((num_rotations, 4))
+        result[:, 3] = 1
+
+        for idx in range(len(seq)):
+            if intrinsic:
+                result = cls._compose_quat(quat[:, :, idx], result)
+            else:
+                result = cls._compose_quat(result, quat[:, :, idx])
+        return result
+
+    @classmethod
+    def from_euler(cls, seq, angles, degrees=False):
+        """Initialize rotation from Euler angles.
+
+        Parameters
+        ----------
+        seq : str
+            `seq` must be a string of upto 3 characters belonging to the set
+            {'x', 'y', 'z'} or {'X', 'Y', 'Z'}. Each lowercase character
+            represents an extrinsic rotation around the corresponding axis and
+            an uppercase character represents an intrinsic rotation. Extrinsic
+            and intrinsic rotations cannot be mixed in one function call.
+        angles : float or array_like, shape (N, ) or (N, [1 or 2 or 3])
+            Euler angles specified in radians (`degrees` is False) or degrees
+            if parameter `degrees` is True.
+            For a single character `seq`, `angles` can be:
+
+                - a single value
+                - array_like, 1D with shape (N, ), where each `angle[i]`
+                  corresponds to a single rotation
+                - array_like, 2D with shape (N, 1), where each `angle[i, 0]`
+                  corresponds to a single rotation
+
+            For 2- and 3-character wide `seq`, `angles` can be:
+
+                - array_like, 1D with shape (W, ) where `W` is the width of
+                  `seq`, which corresponds to a single rotation with `W` axes
+                - array_like, 2D with shape (N, W) with each `angle[i]`
+                  corresponding to a sequence of Euler angles describing a
+                  single rotation
+
+        degrees : boolean
+            If True, then the given angles are taken to be in degrees
+        """
+
+        if len(seq) < 1 or len(seq) > 3:
+            raise ValueError("Expected axis specification to be a non-empty "
+                             "string of upto 3 characters, got {}".format(seq))
+
+        if ((len(seq) == 2 and seq[0] == seq[1]) or
+                (len(seq) == 3 and (seq[0] == seq[1] or seq[1] == seq[2]))):
+            raise ValueError("Expected consecutive axes to be different, "
+                             "got {}".format(seq))
+
+        intrinsic = (recompile('^[XYZ]{1,3}$').match(seq) is not None)
+        extrinsic = (recompile('^[xyz]{1,3}$').match(seq) is not None)
+        if not (intrinsic or extrinsic):
+            raise ValueError("Expected either intrinsic or extrinsic rotations"
```

Comment:
I suggest to be more specific --- "Expected axes from `seq` to be from ['x', 'y', 'z'] or ['X', 'Y, 'Z'], got ...". Or something like that.

Reply:
I think after all checks are done and intrinsic flag is set we can do `seq = seq.lower()`.